src/elements/DipEdge.H

```diff
@@ -104,6 +138,34 @@ namespace impactx::elements
                 * m_g * m_K2 / powi<2>(m_rc);
 
             m_R43 = vf - m_R21;
+
+            // constants for nonlinear map (model = 1):
+
+            // access reference particle values to find beta
+            m_beta = refpart.beta();
+
+            // expressions from eq (14) of Mitchell and Hwang, NAPAC2025
+            amrex::ParticleReal const tan_psi = sin_psi/cos_psi;
+            amrex::ParticleReal const sec_psi = 1_prt/cos_psi;
+
+            // the scale constant specifying entry or exit
+            amrex::ParticleReal const flg = static_cast<amrex::ParticleReal>(m_location);
```

Comment:
Probably here and below `flg` -> `lo(cation)`
```suggestion
            amrex::ParticleReal const loc = static_cast<amrex::ParticleReal>(m_location);
```